### PR TITLE
Small compliance fixes

### PR DESCRIPTION
**Pull request by SquidDev**
I was hoping to get a few more fixes in before opening this. Turns out Markdown's link parsing is a little more awkward than I expected, so felt it was worth opening this before I took a crack at that.

This fixes the following tests:

- 164
  ```
  [Foo bar]:
  <my url>
  'title'
  
  [Foo bar]
  .
  <p><a href="my%20url" title="title">Foo bar</a></p>
  ```
 - 486
   ```
   [link](</my uri>)
   .
   <p><a href="/my%20uri">link</a></p>
   ```
 - 570
   ```
   ![foo ![bar](/url)](/url2)
   .
   <p><img src="/url2" alt="foo bar" /></p>
   ```
 - 591
   ```
   <http://foo.bar.baz/test?q=hello&id=22&boolean>
   .
   <p><a href="http://foo.bar.baz/test?q=hello&amp;id=22&amp;boolean">http://foo.bar.baz/test?q=hello&amp;id=22&amp;boolean</a></p>
   ```

2edf48533d44fd4eff30e89a227c61635658a540 doesn't really change anything. It just made the main inline parser a little easier to read for me - happy to revert if people don't like it.

### CODE REVIEW

**Comment by shonfeder** [comment]
Thanks, @SquidDev! The general shape of things looks good to me, but I'd like to be sure this won't conflict with @sonologico's work on revising the Omd AST before we consider merging.

Either way, I should have time for a proper review later this week.

**Comment by sonologico** [comment]
Thanks for taking the time to make this PR. Taking a look at it today.

**Review by sonologico — COMMENTED** [review]
Looks good to me. The only change that I see as needed is the missing line.

The small refactoring you did also looks good. In general, I would have put it in a separate PR since it was unrelated. Regardless, thanks for doing it.

File: src/parser.ml [hunk 1]
```diff
@@ -1569,19 +1569,67 @@ let inline_link =
     (pair link_destination (option None (ws1 >>> some link_title)))
   <<< ws <<< char ')'
 
+let get_buf buf =
+  let s = Buffer.contents buf in
+  Buffer.clear buf;
+  s
+let text buf acc =
```
[the remark this answers is not in the document]
Comment (SquidDev) [reply src/parser.ml:1576]: It it worth me just running `ocp-indent` on `src/*.ml`? There's a couple of other formatting inconsistencies which may be worth fixing.

[the remark this answers is not in the document]
Comment (sonologico) [reply src/parser.ml:1576]: I thought about that as well or maybe adopting ocamlformat, but I'd rather have that as a separate discusssion/PR.

**Review by sonologico — APPROVED** [review]
Approved. Thank you. I'll merge it soon.

**Pull request merged**